src/main/java/com/w6/nlp/Parser.java

```diff
@@ -52,7 +55,33 @@ else if (parent.label().value().startsWith("N"))
         
         where = LocationParser.parseLocationFromString(input);
         
+        for(String str:where){
+            text.add(new Word(str, "where"));
+        }
+        
         when = DateTimeParser.parseDateAndTimeFromString(input);
+        
+        for(String str:when){
+            text.add(new Word(str, "when"));
+        }
+        
+        what = violentVerbsParser.getAllViolentVerbs(input);
+        
+        for(String str:what){
+            text.add(new Word(str, "what"));
+        }
+        
+        GetDoerAndVictim.ObjectsAndSubjects objAndSubj = GetDoerAndVictim.getSubjectAndObjectOfViolence(input);
+        
+        for(String str:objAndSubj.subjects){
```

Comment:
Did you test it? It will not make sense, because text will not have it's original structure.